src/object_based/htm.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+#include <Eigen/Core>
+#include <array>
+#include <vector>
+
+namespace ear {
+  /// Represents the subdivision of a sphere into a Hierarchical Mesh of
+  /// Triangles; see http://www.skyserver.org/HTM/
+  ///
+  /// This doesn't really do anything except hold the structure, as we don't
+  /// need the standard querying operations for our application (the polar
+  /// extent panner).
+  ///
+  /// Each edge and point has an index, which is shared between triangles.
+  class HTM {
```

Comment:
This one is personal preference, but I'm not a huge fan of classes named after an acronym. Being all upper case can make instantiations look like macro calls and they don't mean much if you don't know the domain well. If HeirarchicalTriangularMesh is too verbose, maybe just HTMesh would give a bit of a clue?